File: String/524LongestWordinDictionarythroughDeleting.cpp
class Solution {
public:
    string findLongestWord(string s, vector<string>& d) {
        string rev;
        int m=0, n=0;
        for(auto v:d){
            for(int m=0, n=0; m<s.size() && n<v.size(); m++){
                if(s[m]==v[n]) n++;
                if(n==v.size() && (rev.size()<v.size() || (rev.size()==v.size()) && v<rev)) rev = v;
            }
        }
        return rev;
    }
};

File: Sort/524longestWordInDictionaryThroughDeleting.py
class Solution(object):
    def findLongestWord(self, s, d):
        """
        :type s: str
        :type d: List[str]
        :rtype: str
        """
        #myc
        def isValid(s,target):
            i, j = 0, 0
            while i < len(s) and j < len(target):
                if s[i] == target[j]:
                    j += 1
                i += 1
            return  j == len(target)
        
        longest = ''
        for i in d:
            l1, l2 = len(longest), len(i)
            if l1 > l2 or (l1 == l2 and ord(longest[0]) < ord(i[0])):
                continue
            if isValid(s,i):
                longest = i
        return longest